File: 백준/Bronze/3234. LUKA/LUKA.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    X, Y = map(int, input().split())
    K = int(input())
    S = input()
    lst = []
    x, y = 0, 0
    if abs(X-x) <= 1 and abs(Y-y) <= 1:
        lst.append(0)
    for k in range(K):
        if S[k] == 'I':
            x += 1
        elif S[k] == 'S':
            y += 1
        elif S[k] == 'Z':
            x -= 1
        else:
            y -= 1
        if abs(X - x) <= 1 and abs(Y - y) <= 1:
            lst.append(k+1)
    if len(lst) == 0:
        print(-1)
    else:
        for l in lst:
            print(l)


if __name__ == '__main__':
    main()

File: 백준/Bronze/3234. LUKA/LUKA.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    int X, Y, K;
    char S[100001];
    int x = 0, y = 0;
    int lst[100001];
    int lst_size = 0;
    scanf("%d %d", &X, &Y);
    scanf("%d", &K);
    getchar();  
    fgets(S, sizeof(S), stdin);
    if (abs(X - x) <= 1 && abs(Y - y) <= 1) {
        lst[lst_size++] = 0;
    }
    for (int k = 0; k < K; k++) {
        char c = S[k];
        if (c == 'I') {
            x += 1;
        } else if (c == 'S') {
            y += 1;
        } else if (c == 'Z') {
            x -= 1;
        } else {
            y -= 1;
        }
        if (abs(X - x) <= 1 && abs(Y - y) <= 1) {
            lst[lst_size++] = k + 1;
        }
    }
    if (lst_size == 0) {
        printf("-1\n");
    } else {
        for (int i = 0; i < lst_size; i++) {
            printf("%d\n", lst[i]);
        }
    }
    return 0;
}